elixir/apps/web/lib/web/components/page_components.ex

```diff
@@ -3,123 +3,45 @@ defmodule Web.PageComponents do
-    attr :navigate, :string
-    attr :icon, :string
-  end
-
-  slot :content, required: true, doc: "A slot for content which lists the entries" do
+  slot :content, required: true, doc: "A slot for content of the section" do
     attr :flash, :any, doc: "The flash to be displayed above the content"
   end
 
-  slot :danger_zone,
-    required: false,
-    doc: "A slot for dangerous actions to be displayed below the content"
-
-  def page(assigns) do
+  def section(assigns) do
     ~H"""
-    <.header>
-      <:title>
-        <%= render_slot(@title) %>
-      </:title>
+    <div class="bg-white dark:bg-gray-800 overflow-hidden border-solid border-slate-200 border-t">
+      <.header>
+        <:title>
+          <%= render_slot(@title) %>
+        </:title>
+
+        <:actions :if={not Enum.empty?(@action)}>
+          <%= for action <- @action do %>
+            <%= render_slot(action) %>
+          <% end %>
+        </:actions>
+      </.header>
 
-      <:actions :if={not Enum.empty?(@action)}>
-        <.action_button
-          :for={action <- @action}
-          type={Map.get(action, :type)}
-          navigate={action.navigate}
-          icon={action.icon}
-        >
-          <%= render_slot(action) %>
-        </.action_button>
-      </:actions>
-    </.header>
-
-    <div class="bg-white dark:bg-gray-800 overflow-hidden">
       <section :for={content <- @content}>
         <div :if={Map.get(content, :flash)}>
           <.flash kind={:info} flash={Map.get(content, :flash)} style="wide" />
           <.flash kind={:error} flash={Map.get(content, :flash)} style="wide" />
         </div>
-
         <%= render_slot(content) %>
       </section>
-
-      <section :if={not Enum.empty?(@danger_zone)}>
-        <.header>
-          <:title>
-            Danger zone
-          </:title>
-          <:actions>
-            <%= render_slot(@danger_zone) %>
-          </:actions>
-        </.header>
-      </section>
     </div>
     """
   end
 
-  attr :navigate, :string,
-    required: false,
-    doc: """
-    The path to navigate to, when set an <a> tag will be used,
-    otherwise a <button> tag will be used
-    """
-
-  attr :type, :string, default: nil, doc: "The style of the button"
-  attr :icon, :string, required: true, doc: "The icon to be displayed on the button"
-  attr :rest, :global
-  slot :inner_block, required: true, doc: "The label for the button"
-
-  def action_button(%{navigate: _} = assigns) do
-    ~H"""
-    <.link class={button_style(@type)} navigate={@navigate} {@rest}>
-      <.icon :if={@icon} name={@icon} class="h-3.5 w-3.5 mr-2" />
-      <%= render_slot(@inner_block) %>
-    </.link>
-    """
-  end
-
-  def action_button(assigns) do
-    ~H"""
-    <button type="button" class={button_style(@type)} {@rest}>
-      <.icon :if={@icon} name={@icon} class="h-3.5 w-3.5 mr-2" />
-      <%= render_slot(@inner_block) %>
-    </button>
-    """
-  end
-
-  defp button_style do
+  def link_style do
```

Comment:
I think this is because we have a name collision with the default `<.link component />` right? Otherwise, we will just override it and add a default styling along with other options that we might need.

Maybe we fix this by:
1. Renaming component modules from `Web.NavigationComponents` to `Web.Components.Navigation`;
2. Changing `web.exs` from doing `import Web.*Components` to `alias Web.Components`;
3. Then using our own components with a module name: `<.input />` will become `<Form.input />`;
4. Add our own `<Navigation.link />` that uses the default one but adds things that we need on top of that.

This has an additional benefit: once the application grows we won't have naming collisions between our own component modules, eg. we will be able to have `<Search.input />` and `<Form.input />`.

WDYT?

Reply:
> I think this is because we have a name collision with the default <.link component /> right?

Yeah, I knew we had talked about and landed on removing the default `<.link>` component and creating our own, but as I mentioned in another comment, wasn't sure if that was going to make this already large PR a little too large.

I think you're idea of namespacing the different component modules sounds good.

Reply:
How would you feel about me taking the suggested component refactor and making it a separate PR?